### PR TITLE
Add org.gnome.Evolution.Extension.evolution-etesync

### PR DESCRIPTION
**Pull request by daftaupe**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. -> as written further, we've been contacted by the Evolution people
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

Hello,

as discussed and asked by the Evolution folks on https://github.com/flathub/org.gnome.Evolution/issues/122, this is the submission of the EteSync plugin for Evolution on behalf of the EteSync project.

### CODE REVIEW

File: org.gnome.Evolution.Extension.evolution-etesync.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
+{
+	"id": "org.gnome.Evolution.Extension.evolution-etesync",
+	"runtime": "org.gnome.Evolution",
+	"branch": "stable",
+	"runtime-version": "master",
```
Comment (hfiguiere) [inline org.gnome.Evolution.Extension.evolution-etesync.json:5]:
```suggestion
	"runtime-version": "stable",
```

Comment (daftaupe) [reply org.gnome.Evolution.Extension.evolution-etesync.json]: I've pushed a fix for that one.

**Comment by hfiguiere** [comment]
bot build org.gnome.Evolution.Extension.evolution-etesync

**Comment by hfiguiere** [comment]
it needs an appstream file.

**Comment by hfiguiere** [comment]
and `flathub.json` with `skip-icon-check: true`

**Comment by daftaupe** [comment]
These should be fixed now.

**Comment by daftaupe** [comment]
bot, build org.gnome.Evolution.Extension.evolution-etesync

**Comment by daftaupe** [comment]
bot, build org.gnome.Evolution.Extension.evolution-etesync

**Comment by daftaupe** [comment]
bot, build org.gnome.Evolution.Extension.evolution-etesync

File: org.gnome.Evolution.Extension.evolution-etesync.json [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+					"sha256": "ae89773450caf25328825576bc9a675ce885e1c775481130c98f89e18555d3a3"
+				}
+			]
+		},
+
+		{
+			"name": "metainfo",
+			"buildsystem": "simple",
+			"build-commands": [
+				"install -t /app/share/metainfo/ -Dm644 org.gnome.Evolution.Extension.evolution-etesync.metainfo.xml"
```
Comment (hfiguiere) [inline org.gnome.Evolution.Extension.evolution-etesync.json:81]:
```suggestion
				"install -t $FLATPAK_DEST/share/metainfo/ -Dm644 org.gnome.Evolution.Extension.evolution-etesync.metainfo.xml"
```

**Comment by daftaupe** [comment]
bot, build org.gnome.Evolution.Extension.evolution-etesync

**Comment by daftaupe** [comment]
bot, build org.gnome.Evolution.Extension.evolution-etesync

File: org.gnome.Evolution.Extension.evolution-etesync.json (hunk 2 above)
Comment (daftaupe) [reply org.gnome.Evolution.Extension.evolution-etesync.json]: I actually changed it to `$FLATPAK_DEST/share/app-info/xmls/`.
But I don't get why it's looking for a gzipped file. Should I create it myself ?

Comment (hfiguiere) [reply org.gnome.Evolution.Extension.evolution-etesync.json]: you should NOT install ANYTHING in this path. The suggestion I gave IS correct.

File: org.gnome.Evolution.Extension.evolution-etesync.json [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+					"sha256": "ae89773450caf25328825576bc9a675ce885e1c775481130c98f89e18555d3a3"
+				}
+			]
+		},
+
+		{
+			"name": "metainfo",
+			"buildsystem": "simple",
+			"build-commands": [
+				"install -t $FLATPAK_DEST/share/app-info/xmls/ -Dm644 org.gnome.Evolution.Extension.evolution-etesync.metainfo.xml"
```
Comment (hfiguiere) [inline org.gnome.Evolution.Extension.evolution-etesync.json:81]:
```suggestion
				"install -t $FLATPAK_DEST/share/metainfo -Dm644 org.gnome.Evolution.Extension.evolution-etesync.metainfo.xml"
```


you never ever install anything in `app-info`.

Comment (hfiguiere) [reply org.gnome.Evolution.Extension.evolution-etesync.json:81]: you also need to run the appstream-compose by hand like any other extension.

File: org.gnome.Evolution.Extension.evolution-etesync.json (hunk 2 above)
Comment (daftaupe) [reply org.gnome.Evolution.Extension.evolution-etesync.json]: ok, but I followed your suggestion in my previous attempt, and the build also failed.

Comment (hfiguiere) [reply org.gnome.Evolution.Extension.evolution-etesync.json]: https://github.com/flathub/flathub/pull/4297#discussion_r1248815007

File: org.gnome.Evolution.Extension.evolution-etesync.json (hunk 3 above)
Comment (hfiguiere) [reply org.gnome.Evolution.Extension.evolution-etesync.json:81]: `"appstream-compose --basename=${FLATPAK_ID} --prefix=${FLATPAK_DEST} --origin=flatpak ${FLATPAK_ID}"`

**Comment by daftaupe** [comment]
bot, build org.gnome.Evolution.Extension.evolution-etesync

**Comment by daftaupe** [comment]
bot, build org.gnome.Evolution.Extension.evolution-etesync

File: org.gnome.Evolution.Extension.evolution-etesync.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+				}
+			]
+		},
+
+		{
+			"name": "metainfo",
+			"buildsystem": "simple",
+			"build-commands": [
+				"install -t $FLATPAK_DEST/share/metainfo -Dm644 org.gnome.Evolution.Extension.evolution-etesync.metainfo.xml",
+				"appstream-compose --basename=org.gnome.Evolution.Extension.evolution-etesync --prefix=${FLATPAK_DEST} --origin=org.gnome.Evolution.Extension.evolution-etesync"
```
Comment (hfiguiere) [inline org.gnome.Evolution.Extension.evolution-etesync.json:82]:
```suggestion
				"install -t $FLATPAK_DEST/share/metainfo -Dm644 org.gnome.Evolution.Extension.evolution-etesync.metainfo.xml",
				"appstream-compose --basename=org.gnome.Evolution.Extension.evolution-etesync --prefix=${FLATPAK_DEST} --origin=flatpak org.gnome.Evolution.Extension.evolution-etesync"
```

**Comment by daftaupe** [comment]
bot, build org.gnome.Evolution.Extension.evolution-etesync

File: org.gnome.Evolution.Extension.evolution-etesync.metainfo.xml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="runtime">
+  <id>org.gnome.Evolution.Extension.evolution-etesync</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>LGPL</project_license>
```
Comment (hfiguiere) [inline org.gnome.Evolution.Extension.evolution-etesync.metainfo.xml:5]: this is not a valid SPDX identifier.

Comment (hfiguiere) [reply org.gnome.Evolution.Extension.evolution-etesync.metainfo.xml:5]: https://spdx.dev/licenses/

Comment (daftaupe) [reply org.gnome.Evolution.Extension.evolution-etesync.metainfo.xml]: thanks for the pointer

**Comment by daftaupe** [comment]
bot, build org.gnome.Evolution.Extension.evolution-etesync

**Comment by daftaupe** [comment]
Thanks for your help @hfiguiere !
What are the next steps ?

**Pull request closed without merging**